src/utils/wallet.js

```diff
@@ -113,56 +114,56 @@ class Wallet {
         this.twoFactor = new TwoFactor(this)
     }
 
-    async getLocalAccessKey(accountId, accessKeys) {
-        const localPublicKey = await this.inMemorySigner.getPublicKey(accountId, NETWORK_ID)
-        return localPublicKey && accessKeys.find(({ public_key }) => public_key === localPublicKey.toString())
+    isEmpty() {
+        return !this.accounts || !Object.keys(this.accounts).length
     }
 
-    async getLedgerKey(accountId) {
-        // TODO: All callers should specify accountId explicitly
-        accountId = accountId || this.accountId
-        // TODO: Refactor so that every account just stores a flag if it's on Ledger?
-
-        const accessKeys = await this.getAccessKeys(accountId)
-        if (accessKeys) {
-            const localKey = await this.getLocalAccessKey(accountId, accessKeys)
-            const ledgerKey = accessKeys.find(accessKey => accessKey.meta.type === 'ledger')
-            if (ledgerKey && (!localKey || localKey.permission !== 'FullAccess')) {
-                return PublicKey.from(ledgerKey.public_key)
-            }
+    getAccount(accountId = this.accountId) {
+        if (this.has2fa && this.twoFactor.accountId === accountId) {
+            return this.twoFactor;
```

Comment:
@mattlockyer this looks like potentially pretty big problem when switching accounts. `Account` for sure doesn't expect being reused between different accounts. Doesn't seem like `TwoFactor` code has anything for switching accounts either (it takes `accountId` in constructors).

Reply:
Apologies for moving methods.

It made sense to have everything related to account management close to the wallet constructor.

Attempting to simplify wallet.js codebase by having TwoFactor extend Account.

When switching accounts, `refreshAccount` is called, and the instance of `twoFactor`, property (child) of wallet, is re-initialized.

https://github.com/near/near-wallet/blob/1c088935cad80500eddca4f17e1b39d9009e214f/src/utils/wallet.js#L121-L173